fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/source/enumerator/FlinkSourceEnumerator.java

```diff
@@ -263,6 +271,35 @@ private Set<PartitionInfo> listPartitions() {
         }
     }
 
+    /** Apply partition filter. */
+    private List<PartitionInfo> applyPartitionFilter(List<PartitionInfo> partitionInfos) {
+        if (partitionFilters != null && !partitionFilters.isEmpty()) {
+            return partitionInfos.stream()
+                    .filter(
+                            partitionInfo -> {
+                                Map<String, String> specMap =
+                                        partitionInfo.getPartitionSpec().getSpecMap();
+                                for (PushdownUtils.FieldEqual filter : partitionFilters) {
+                                    String partitionValue =
+                                            specMap.get(
+                                                    tableInfo
```

Comment:
Move  `tableInfo.getRowType().getFieldNames()` out of for loop to reduce redundant call